src/cljs/org/broadinstitute/firecloud_ui/page/library/library_page.cljs

```diff
@@ -130,15 +130,40 @@
                   (callback {:error status-text})))})))))})
 
 (react/defc SearchSection
-  {:render
+  {:get-filters
    (fn [{:keys [props]}]
+     (utils/map-keys name (:facet-filters props)))
+   :render
+   (fn [{:keys [props this]}]
      [:div {}
       [:div {:style {:fontWeight 700 :fontSize "125%" :marginBottom "1em"}} "Search Filters:"]
       [:div {:style {:background (:background-light style/colors) :padding "16px 12px"}}
-       [comps/TextFilter {:ref "text-filter"
-                          :initial-text (:search-text props)
-                          :width "100%" :placeholder "Search"
-                          :on-filter (:on-filter props)}]]])})
+       [comps/AutocompleteFilter
+        {:on-filter (:on-filter props)
+         :width "100%"
+         :initial-text (:search-text props)
+         :ref "text-filter"
+         :placeholder "Search"
+         :facet-filters (:facet-filters props)
+         :bloodhoundInfo {:url (str (config/api-url-root) "/api/library/suggest")
+                          :transform (fn [response]
+                                       (clj->js (mapv (partial hash-map :value) (aget response "results"))))
+                          :cache false
+                          :prepare (fn [query settings]
+                                      (clj->js
+                                        (assoc (js->clj settings)
+                                          :headers {:Authorization (str "Bearer " (utils/get-access-token))}
+                                          :type "POST"
+                                          :contentType "application/json; charset=UTF-8"
+                                          :data (utils/->json-string
+                                                  {:searchString query
+                                                   :filters (react/call :get-filters this)
+                                                   :from 0
+                                                   :size 10}))))}
+         :typeaheadDisplay (fn [result]
+                             (clojure.string/replace (aget result "value") #"<strong class='es-highlight'>|</strong>" ""))
```

Comment:
jquery has a `text()` method that handles this in a pretty generic way, if you feel like reaching out to jquery. Plain JavaScript has `innerText`, too - but that would require some runaround, because you'd have to put the result inside a DOM element in order to get that element's text.

summary: I find this code kinda ugly, but I don't have terrific suggestions to do it better.